### PR TITLE
Allow to delete passwords on public links when password is enforced but permission is set

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
We've enabled to ability to allow delete passwords on public links, even if the password is enforced.
Therefore, the user needs respective permission, granted by the server.
This feature is only possible on public links that have the viewer role.

## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/7538

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by JanAckermann** [comment]
issues: 
- password banned list errors not shown in frontent
- links will be crated again instead of copied

File: packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue [hunk 1]
```diff
@@ -422,7 +445,7 @@ export default defineComponent({
     checkLinkToUpdate({ link }) {
       const params = this.getParamsForLink(link)
 
-      if (!link.password && this.isPasswordEnforcedFor(link)) {
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue:425]: check again with fresh mind

Comment (JanAckermann) [reply packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue:425]: I think that's ok

File: packages/web-pkg/src/quickActions.ts [hunk 2]
```diff
@@ -74,28 +74,9 @@ export default {
...
-              clientService,
-              language,
-              password,
-              resource: item,
-              store
-            })
-          }
-        )
-      }
-
```
Comment (JanAckermann) [inline packages/web-pkg/src/quickActions.ts:96]: centralized

File: packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue [hunk 3]
```diff
@@ -338,7 +342,7 @@ export default defineComponent({
           })
         }
       }
-      if (!this.isPasswordEnforced && !this.link.password && !this.isAliasLink) {
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue:341]: Huh I don't get this check in the first place? If password is not enforced we don't allow the user to add a password ?

**Review by JammingBen — APPROVED** [review]
👍

**Pull request merged**